src/target/riscv/riscv.c

```diff
@@ -527,6 +527,7 @@ static int maybe_add_trigger_t2(struct target *target,
 	}
 
 	/* address/data match trigger */
+	tdata1 = set_field(tdata1, CSR_MCONTROL_TYPE(riscv_xlen(target)), 2);
```

Comment:
This looks as a correct fix (we should always try to write the desired value to `tdata1.type` since it is not known whether that field is read-only or not).